homework-4/classes/CollectionDictionary.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CollectionApplication
+{
+    class CollectionDictionary
+    {
+        private Dictionary<string, int> elements = new Dictionary<string, int>();
+
+        public void Menu()
+        {
+            while (true)
+            {
+                Console.WriteLine("\nPlease enter a number from the given menu:");
+                Console.WriteLine("1. Add an element to the Dictionary.");
+                Console.WriteLine("2. Print all the elements present in the Dictionary.");
+                Console.WriteLine("3. Delete the last element from the Dictionary.");
+                Console.WriteLine("4. Delete the first element from the Dictionary.");
+                Console.WriteLine("5. Delete the middle element from the Dictionary.");
+                Console.WriteLine("6. Calculate the average of the elements present in the Dictionary.");
+                Console.WriteLine("7. Go back to the previous menu.");
+                Console.WriteLine("8. Exit the application.");
+
+                if (!int.TryParse(Console.ReadLine(), out int choice))
+                {
+                    Console.WriteLine("Please enter a valid number.");
+                    continue;
+                }
+
+                switch (choice)
+                {
+                    case 1:
+                        AddElement();
+                        break;
+                    case 2:
+                        PrintElements();
+                        break;
+                    case 3:
+                        DeleteLastElement();
+                        break;
+                    case 4:
+                        DeleteFirstElement();
+                        break;
+                    case 5:
+                        DeleteMiddleElement();
+                        break;
+                    case 6:
+                        CalculateAverage();
+                        break;
+                    case 7:
+                        return;
+                    case 8:
+                        Console.WriteLine("Exiting the application...");
+                        Environment.Exit(0);
+                        break;
+                    default:
+                        Console.WriteLine("Please enter a valid number.");
+                        break;
+                }
+            }
+        }
+
+        private void AddElement()
+        {
+            Console.WriteLine("Enter the key:");
+            string key = Console.ReadLine();
```

Comment:
Readline reads the next line of characters from the standard input stream and returns the next line of characters from the input stream, or null if no more lines are available.

so here key can be null and you should handle it here like--
```
string key = Console.ReadLine() ?? "";
if(key.Length == 0)
{
     Console.WriteLine("Please enter a valid non empty key.");
}
```